pkg/subctl/cmd/root.go

```diff
@@ -234,3 +240,27 @@ var nodeLabelBackoff wait.Backoff = wait.Backoff{
 	Factor:   1.2,
 	Jitter:   1,
 }
+
+func checkVersionMismatch(cmd *cobra.Command, args []string) error {
+	config, err := getRestConfig(kubeConfig, kubeContext)
+	exitOnError("The provided kubeconfig is invalid", err)
+
+	clientSet, err := submarinerclientset.NewForConfig(config)
+	exitOnError("Unable to get the Submariner config", err)
+
+	submariner, err := clientSet.SubmarinerV1alpha1().Submariners(OperatorNamespace).Get(submarinercr.SubmarinerName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	if submariner != nil && submariner.Spec.Version != "" {
```

Comment:
Should we react (error/warning) if we don't find the expected resource/versions, instead of basically doing the same path as the a successful match? Same applies to the conditional below.

Reply:
We shouldn't I believe, otherwise you can't deploy-broker, or join a cluster :-)

The show command is ready for submariner not being installed.

Reply:
I'm not sure if the `Spec.Version` is mandatory.